### ISSUE TITLE
Task we can address based on the embeddings

### ISSUE THREAD

**Issue by HelloWorldLTY** [snap-stanford/UCE#7]
Hi, I wonder what types of tasks we can address based on the current embeddings. It seems that in the original paper UCE has multiple functions, but I am not sure what types of tasks need fine-tuning. Thanks.

**Comment by Yanay1** [comment]
The UCE model is not fine tuned for any tasks-- the core embedding model is always zero shot.

Given one dataset, you can do a variety of tasks like clustering cell types or inferring hierarchies (using default scanpy functions for example).

Given embeddings of multiple datasets, you can do additional tasks like transferring labels using simple models (we used logistic classifier from SK Learn), model free transfer like using nearest neighbors or centroids based methods, plus the other single dataset tasks in a multi dataset context.

There are many other possible downstream tasks. As an example, for any model which would have previously used gene expression as a representation of a cell, you can replace the gene expression with UCE, train the model, and then apply it to other UCE embedded datasets. 

For mutli dataset tasks, we replaced a large sample of the IMA which you can download from here: https://github.com/snap-stanford/UCE#data

**Comment by HelloWorldLTY** [comment]
Hi, got it, thanks for your quick feedback.

Is it possible for me to access your codes for using the logistic regression classifier to do cell type annotation? I intend to reproduce this process with unbiased design. Thanks.

**Comment by Yanay1** [comment]
Is there a specific result or dataset from the paper you want to replicate?

The method is the same as is listed in SK Learn documentation, for example:

```
from sklearn.linear_model import LogisticRegression
X = reference_ad.obsm["X_uce"] # for some uploaded version of IMA, the UCE embeds are just in .X, but others are in .obsm["X_uce"]

y = reference_ad.obs["..."] # some metadata label like cell type

clf = LogisticRegression(random_state=0).fit(X, y)

predicted_y = clf.predict(transfer_ad.obsm["X_uce"])

```

**Comment by HelloWorldLTY** [comment]
Thanks a lot. I ask this question because logisticregression also has its own hyper-parameters.

**Comment by Yanay1** [comment]
Ah that makes sense! For all uses of logistic regression / classification, the default parameters for sklearn were used.
